### DISCUSSION TITLE
Why use "Composable" annotation for a function that doesn't describe the UI ?

### DISCUSSION THREAD

**Discussion by florian-mlr** [android/nowinandroid#778] [Q&A]
In this project, I notice some functions that are not intended to describe the UI or not retaining the state like the following one :

```
NiaTheme(
   androidTheme = shouldUseAndroidTheme(uiState),
) {....}

@Composable
private fun shouldUseAndroidTheme(
    uiState: MainActivityUiState,
): Boolean = when (uiState) {
    Loading -> false
    is Success -> when (uiState.userData.themeBrand) {
        ThemeBrand.DEFAULT -> false
        ThemeBrand.ANDROID -> true
    }
}
```

What is the point of the Compose annotation in this case ? Is it a "good practice" ?

**Comment by thagikura** [comment]
It's not necessary. I think they were left unintentionally.

**Comment by myofficework000** [comment]
In general, it's recommended to follow standard conventions and best practices when using annotations. If you're not working with Jetpack Compose or UI-related code, it's best to omit the @Composable annotation to avoid confusion and make your codebase more maintainable.

https://medium.com/@myofficework000/revolutionizing-android-ui-development-with-jetpack-compose-a-comprehensive-guide-af86026267af
